lineapy/graph_reader/apis.py

```diff
@@ -27,6 +27,7 @@
     GetVersionEvent,
     track,
 )
+from lineapy.utils.deprecation_utils import lru_cache
```

Comment:
only do this if lru_cache isnt already available. The newer versions of python use cpython and other thread safe mechanism that might not be as efficient as the pure python implementation here.

```suggestion
try:
    from functools import lru_cache  # type: ignore
except ImportError:  # pragma: no cover
    from lineapy.utils.deprecation_utils import lru_cache # type: ignore
```

The type ignores might be needed if your mypy is bugging you because the types of inbuilt lru cache and your replacement might parse to diff types.